### PR TITLE
refactor: replace fungible token example sim tests with workspaces

### PR DESCRIPTION
**Pull request by itegulov**
I had to cut a few asserts from some of the tests since workspaces does not expose some of the stuff they needed, but at the very least all of the tests have survived the migration without losing meaningfulness. I left TODOs in appropriate places for us to revisit later.

Also, I took the liberty to delete `examples/heavy.rs` since it was originally added in #307 to showcase sim tests performance. But lmk if you would prefer for me to keep and refactor it too.

Related to #563 #662

### CODE REVIEW

File: examples/fungible-token/tests/workspaces.rs
Comment (itegulov) [inline examples/fungible-token/tests/workspaces.rs:24]: @austinabell is there a way to instantiate `None` with an explicit generic type in one line? I have tried `None: Option<bool>`, but apparently type ascriptions are experimental.

**Review by austinabell — COMMENTED** [review]
oops forgot to finalize this review, sorry if any are stale comments. Was just a quick scan until the CI passes, which seemed more workspaces/sandbox utils related

File: examples/fungible-token/tests/workspaces.rs
Comment (itegulov) [reply examples/fungible-token/tests/workspaces.rs]: Yeah, at first I have refactored these values to look like `10000 * ONE_NEAR`, but later I found out about `parse_near!` so ended up with in-place `parse_near!("10000 N")` instead. lmk which way you prefer.

Comment (itegulov) [reply examples/fungible-token/tests/workspaces.rs]: Yeah, at first I have refactored these values to look like `10000 * ONE_NEAR`, but later I found out about `parse_near!` so ended up with in-place `parse_near!("10000 N")` instead. lmk which way you prefer.

File: examples/fungible-token/tests/workspaces.rs
Comment (itegulov) [reply examples/fungible-token/tests/workspaces.rs:214]: I agree, that would simplify user's life

File: examples/fungible-token/tests/workspaces.rs
Comment (itegulov) [reply examples/fungible-token/tests/workspaces.rs]: Yep, already figured this out with @ChaoticTempest's help

File: examples/fungible-token/tests/workspaces.rs
Comment (itegulov) [reply examples/fungible-token/tests/workspaces.rs:17]: The problem with `()` is that it does not match the function's signature. Even if `None` and `()` do serialize the same way, as an external user reading these examples I would be confused why we are passing `()` instead of `Option<bool>`.

`Option::<bool>::None` works though, thanks!

File: examples/fungible-token/tests/workspaces.rs
Comment (itegulov) [reply examples/fungible-token/tests/workspaces.rs:214]: I agree, that would simplify my life as a workspaces user

**Comment by itegulov** [comment]
> oops forgot to finalize this review, sorry if any are stale comments.

No worries. Indeed, I have already refactored most of the weird stuff away.

> Was just a quick scan until the CI passes, which seemed more workspaces/sandbox utils related

Yeah, I am still trying to figure out why Github Actions don't want to run the downloaded sandbox binaries... There is some weirdness going on in Buildkite too, but I have not looked too deeply into that yet.

File: examples/fungible-token/tests/workspaces.rs
Comment (ChaoticTempest) [reply examples/fungible-token/tests/workspaces.rs:213]: Interesting. Normally the pattern for these was to do an `into_result()?` to forward the error, but I see that in this case we want to do assertions to be more explicit. Good idea overall, I'll add a ticket

File: examples/fungible-token/tests/workspaces.rs
Comment (ChaoticTempest) [inline examples/fungible-token/tests/workspaces.rs:417]: It's not necessary to have the `as_account()` to retrieve the id. You can simply just do `contract.id()` for these

File: examples/fungible-token/tests/workspaces.rs
Comment (itegulov) [reply examples/fungible-token/tests/workspaces.rs:59]: indeed, thanks

File: examples/fungible-token/tests/workspaces.rs
Comment (itegulov) [reply examples/fungible-token/tests/workspaces.rs:417]: good point!

File: examples/fungible-token/tests/workspaces.rs
Comment (itegulov) [reply examples/fungible-token/tests/workspaces.rs:205]: Well, `CallExecutionDetails` does not have `into_result()` anyway. Thanks for the PR, I will take a look shortly

File: examples/fungible-token/tests/workspaces.rs
Comment (austinabell) [inline examples/fungible-token/tests/workspaces.rs:197]: can you explain what you mean by this? Not completely intuitive

**Review by austinabell — APPROVED** [review]
LGTM, I'll wait for @ChaoticTempest approval before pulling in because he might have some tricks :)

**Review by ChaoticTempest — APPROVED** [review]
looks good to me as well. Will be adding the promise results soon, so hopefully we can add in those commented out `promise_error` checks

File: examples/fungible-token/Cargo.toml
Comment (austinabell) [inline examples/fungible-token/Cargo.toml:9]: yeah an extension of what Phuong mentioned, this shouldn't be included
```suggestion
```

File: examples/fungible-token/tests/workspaces.rs
Comment (itegulov) [reply examples/fungible-token/tests/workspaces.rs:197]: yeah sorry for using incorrect terminology. What I meant to say is that these two actions (fn call `ft_transfer_call` and fn call `storage_unregister`) are supposed to be [batched](https://nomicon.io/RuntimeSpec/Transactions#batched-transaction) into a single transaction according to the old simtest-powered tests. Since workspaces do not support batched transactions yet, I had to split these two actions into two separate transactions. But to be fair I do not really see why you would want to batch these two specific action other than to (maybe?) save some gas... Atomicity does not matter here I think.

File: examples/fungible-token/Cargo.toml
Comment (itegulov) [reply examples/fungible-token/Cargo.toml:9]: As @ChaoticTempest mentioned in the other thread I actually need this to pull `FinalExecutionStatus`

Comment (austinabell) [reply examples/fungible-token/Cargo.toml:9]: This seems like something that shouldn't be exposed through workspaces

Comment (itegulov) [reply examples/fungible-token/Cargo.toml:9]: Hmm, do you mean that workspaces should have its own type for expressing the call execution status?

CC @ChaoticTempest

Comment (austinabell) [reply examples/fungible-token/Cargo.toml:9]: yeah, this is not an ideal pattern for workspace users to have to import a nearcore internal crate. Not a big deal, but something that should be changed at some point

Comment (ChaoticTempest) [reply examples/fungible-token/Cargo.toml:9]: ooof, this was a pattern I wasn't expecting to cover with `asserts!` with `FinalExecutionStatus`. Think we can defer this later until a new minor version of workspaces is released and we can convert these `asserts!` over to using `is_success` calls

Comment (itegulov) [reply examples/fungible-token/Cargo.toml:9]: > Think we can defer this later until a new minor version of workspaces is released and we can convert these `asserts!` over to using `is_success` calls

Yeah, we can defer for sure and `is_success` will help in this case, but `CallExecutionDetails::status` (which is a part of the public API) will still be a type from `near-primitives` so I think eventually we would want to replace it with something else.

**Pull request merged**
